### PR TITLE
[server] Cleanup unused data (comments, review statuses)

### LINKED ISSUE #3142: Review status and comment garbage collection
**Describe the bug**
When I remove a run, the corresponding review status and comments are not cleared from the database. This is understandable, since other runs may contain bug hashes to these comments and review statuses. However, we should consider removing these fields in a garbage collection phase at server start, otherwise it may be surprising that these statuses and comments reappear after storing the run again.

**CodeChecker version**
6.15.0

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #3142

### CODE REVIEW

**Review by dkrupp — COMMENTED** [review]
Does this cleanup job runs when there is a run removal, right?

**Comment by csordasmarton** [comment]
@dkrupp Yes, it will run when the user removes a run or when the web server is started (similar to unused files).

File: web/server/codechecker_server/api/report_server.py [hunk 1]
```diff
@@ -2442,9 +2442,8 @@ def removeRunReports(self, run_ids, report_filter, cmp_data):
                 LOG.error(ex)
                 return False
 
-        # Delete files and contents that are not present
-        # in any bug paths.
-        db_cleanup.remove_unused_files(self.__Session)
+        # Remove unused data (files, comments etc) from the database.
```
Comment (whisperity) [inline web/server/codechecker_server/api/report_server.py:2445]:
```suggestion
        # Remove unused data (files, comments, etc.) from the database.
```

File: web/server/codechecker_server/database/db_cleanup.py [hunk 2]
```diff
@@ -79,6 +79,57 @@ def remove_unused_files(session_maker):
...
+def remove_unused_data(session_maker):
+    """ Remove unused data (files, comments, etc.) from the database. """
+    remove_unused_files(session_maker)
+    remove_unused_comments(session_maker)
+    remove_unused_review_statuses(session_maker)
+
+
+def remove_unused_comments(session_maker):
+    """ Remove unused comments from the database. """
+    LOG.debug("Garbage collection of unused comments started...")
```
Comment (whisperity) [inline web/server/codechecker_server/database/db_cleanup.py:91]: What's an "unused comment"? Perhaps it'd be better to call it dangling.

**Review by whisperity — COMMENTED** [review]
What happens if a bug is reopened/reintroduced later? Will we lose all the comments that were previously posted to such a bug report?

**Review by bruntib — APPROVED** [review]

**Pull request merged**